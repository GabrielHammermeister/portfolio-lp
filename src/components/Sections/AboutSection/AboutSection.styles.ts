import { Link } from 'components/IconLink/IconLink.styles'
import styled from 'styled-components'

export const Container = styled.main`
   z-index: 2;
   color: white;
   display: flex;
   flex-direction: column;
   align-items: center;
   height: 100vh;
   margin-bottom: 20vh;
   padding: 0 50px;
   gap: 55px;

   @media (min-width: 900px) {
      flex-direction: row-reverse;
      align-items: flex-start;
      justify-content: space-between;
   }

   @media (min-width: 500px) {
      padding: 0 100px;
   }
   @media (min-width: 900px) {
      padding: 0 150px;
   }
`
export const Illustration = styled.img`
   width: clamp(300px, 50%, 800px);
   padding-top: 6rem;
`

export const Content = styled.aside`
   flex: 1;
   text-align: center;
   & h2 {
      margin-bottom: 1.4rem;
   }

   & p {
      margin-bottom: 3rem;
      overflow: hidden;
      text-overflow: ellipsis;
      line-clamp: 4;
      display: -webkit-box;
      -webkit-line-clamp: 4;
      -webkit-box-orient: vertical;
   }

   @media (max-width: 900px) {
      & ${Link} {
         display: none;
      }
   }

   @media (min-width: 900px) {
      text-align: left;
      padding-top: 6.25rem;
      display: flex;
      flex-direction: column;
      max-width: 900px;
   }
`

export const LinksContainer = styled.div`
   color: ${({ theme }) => theme.text.dark};
   font-size: 32px;
   display: flex;
   align-items: center;
   justify-content: center;
   gap: 30px;

   @media (min-width: 900px) {
      justify-content: flex-start;
   }
`

export const Button = styled.button`
   color: ${({ theme }) => theme.text.light};
   border: none;
   padding: 10px 20px;
   border-radius: 2px;
   height: 47px;
   background: ${({ theme }) => theme.text.primary};

   font-size: 20px;
   font-weight: 500;
   line-height: 29px;
`
