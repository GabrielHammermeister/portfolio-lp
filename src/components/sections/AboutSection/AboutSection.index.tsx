import IconLink from 'components/IconLink/IconLink.index'
import Typography from 'components/Typography/Typography.index'
import { AiFillGithub, AiFillLinkedin } from 'react-icons/ai'
import * as S from './AboutSection.styles'

export const AboutSection = () => {
   return (
      <>
         <S.Container id="About">
            <S.Illustration src="/illustrations/blogging.svg" />
            <S.Content>
               <Typography type="heading-2" color="primary">
                  About Me
               </Typography>
               <Typography type="body" color="dark">
                  An inquisitive Computer Science Engineering student, skilled
                  in leadership, seeking to leverage solid development skills
                  with focus on collaboration, communication and passion.
               </Typography>
               <S.LinksContainer>
                  <S.Button>
                     <Typography type="button" color="light">
                        Download CV
                     </Typography>
                  </S.Button>
                  <IconLink href={'https://github.com/GabrielHammermeister'}>
                     <AiFillGithub />
                  </IconLink>
                  <IconLink
                     href={'https://www.linkedin.com/in/gabriel-hammer/'}
                  >
                     <AiFillLinkedin />
                  </IconLink>
               </S.LinksContainer>
            </S.Content>
         </S.Container>
      </>
   )
}
