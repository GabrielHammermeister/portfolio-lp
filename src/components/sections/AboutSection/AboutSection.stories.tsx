import AboutSection from './AboutSection.index'
import { Meta, Story } from '@storybook/react'

export default {
   title: 'AboutSection',
   component: AboutSection
} as Meta

export const Basic: Story = (args) => <AboutSection {...args} />
