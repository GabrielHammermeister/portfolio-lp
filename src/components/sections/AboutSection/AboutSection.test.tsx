import AboutSection from './AboutSection.index'
import { render, screen } from '@testing-library/react'

describe('<AboutSection />', () => {
   it('should render the heading', () => {
      render(<AboutSection />)

      expect(screen.getByRole('main')).toBeInTheDocument()
   })
})
