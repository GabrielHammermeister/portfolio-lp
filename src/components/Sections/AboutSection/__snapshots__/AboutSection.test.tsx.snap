// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutSection /> should render the heading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c9 {
  color: inherit;
}

.c3 {
  font-size: 2rem;
  color: #F44336;
  line-height: 2.875rem;
  font-weight: 700;
}

.c4 {
  font-size: 1.5rem;
  color: #000C24;
  line-height: 170%;
  font-weight: 400;
}

.c7 {
  font-size: 1rem;
  color: #FBFBFB;
  font-weight: 400;
  line-height: 130%;
}

.c0 {
  z-index: 2;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  margin-bottom: 20vh;
  padding: 0 50px;
  gap: 55px;
}

.c1 {
  width: clamp(300px,50%,800px);
  padding-top: 6rem;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.c2 h2 {
  margin-bottom: 1.4rem;
}

.c2 p {
  margin-bottom: 3rem;
  overflow: hidden;
  text-overflow: ellipsis;
  line-clamp: 4;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
}

.c5 {
  color: #000C24;
  font-size: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 30px;
}

.c6 {
  color: #FBFBFB;
  border: none;
  padding: 10px 20px;
  border-radius: 2px;
  height: 47px;
  background: #F44336;
  font-size: 20px;
  font-weight: 500;
  line-height: 29px;
}

@media (min-width:900px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:500px) {
  .c0 {
    padding: 0 100px;
  }
}

@media (min-width:900px) {
  .c0 {
    padding: 0 150px;
  }
}

@media (max-width:900px) {
  .c2 .c8 {
    display: none;
  }
}

@media (min-width:900px) {
  .c2 {
    text-align: left;
    padding-top: 6.25rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    max-width: 900px;
  }
}

@media (min-width:900px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<body>
    <div>
      <main
        class="c0"
        id="About"
      >
        <img
          class="c1"
          src="/illustrations/blogging.svg"
        />
        <aside
          class="c2"
        >
          <h2
            class="c3"
          >
            About Me
          </h2>
          <p
            class="c4"
          >
            An inquisitive Computer Science Engineering student, skilled in leadership, seeking to leverage solid development skills with focus on collaboration, communication and passion.
          </p>
          <div
            class="c5"
          >
            <button
              class="c6"
            >
              <span
                class="c7"
              >
                Download CV
              </span>
            </button>
            <a
              class="c8 c9"
              href="https://github.com/GabrielHammermeister"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                />
              </svg>
            </a>
            <a
              class="c8 c9"
              href="https://www.linkedin.com/in/gabriel-hammer/"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
                />
              </svg>
            </a>
          </div>
        </aside>
      </main>
    </div>
  </body>,
  "container": .c9 {
  color: inherit;
}

.c3 {
  font-size: 2rem;
  color: #F44336;
  line-height: 2.875rem;
  font-weight: 700;
}

.c4 {
  font-size: 1.5rem;
  color: #000C24;
  line-height: 170%;
  font-weight: 400;
}

.c7 {
  font-size: 1rem;
  color: #FBFBFB;
  font-weight: 400;
  line-height: 130%;
}

.c0 {
  z-index: 2;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  margin-bottom: 20vh;
  padding: 0 50px;
  gap: 55px;
}

.c1 {
  width: clamp(300px,50%,800px);
  padding-top: 6rem;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.c2 h2 {
  margin-bottom: 1.4rem;
}

.c2 p {
  margin-bottom: 3rem;
  overflow: hidden;
  text-overflow: ellipsis;
  line-clamp: 4;
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
}

.c5 {
  color: #000C24;
  font-size: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 30px;
}

.c6 {
  color: #FBFBFB;
  border: none;
  padding: 10px 20px;
  border-radius: 2px;
  height: 47px;
  background: #F44336;
  font-size: 20px;
  font-weight: 500;
  line-height: 29px;
}

@media (min-width:900px) {
  .c0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:500px) {
  .c0 {
    padding: 0 100px;
  }
}

@media (min-width:900px) {
  .c0 {
    padding: 0 150px;
  }
}

@media (max-width:900px) {
  .c2 .c8 {
    display: none;
  }
}

@media (min-width:900px) {
  .c2 {
    text-align: left;
    padding-top: 6.25rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    max-width: 900px;
  }
}

@media (min-width:900px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div>
    <main
      class="c0"
      id="About"
    >
      <img
        class="c1"
        src="/illustrations/blogging.svg"
      />
      <aside
        class="c2"
      >
        <h2
          class="c3"
        >
          About Me
        </h2>
        <p
          class="c4"
        >
          An inquisitive Computer Science Engineering student, skilled in leadership, seeking to leverage solid development skills with focus on collaboration, communication and passion.
        </p>
        <div
          class="c5"
        >
          <button
            class="c6"
          >
            <span
              class="c7"
            >
              Download CV
            </span>
          </button>
          <a
            class="c8 c9"
            href="https://github.com/GabrielHammermeister"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
              />
            </svg>
          </a>
          <a
            class="c8 c9"
            href="https://www.linkedin.com/in/gabriel-hammer/"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1 1 68.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
              />
            </svg>
          </a>
        </div>
      </aside>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`
